### PR TITLE
_terminate_processes for non-docker processes, like nodejs

### PR DESCRIPTION
**Pull request by mr-c**
Fix for the `cwltool` side of https://github.com/DataBiosphere/toil/issues/3965

### CODE REVIEW

**Comment by mr-c** [comment]
@EtiennePer and @w-gao Let me know if this helps

File: cwltool/main.py [hunk 1]
```diff
@@ -132,6 +132,10 @@ def _terminate_processes() -> None:
                         p.kill()
             except FileNotFoundError:
                 pass
+        if process.stdin:
+            process.stdin.close()
+        process.wait(10)
```
Comment (w-gao) [inline cwltool/main.py:137]: My concern is that this wait may raise a [`TimeoutExpired`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait) exception if the process doesn't end within 10s for some reason, which would interrupt the rest of the cleanup.

**Review by w-gao — COMMENTED** [review]
Thank you for fixing this, @mr-c! I tested this with `cwltool` and the nodejs processes are cleaned up now. The next step would be to actually call this function in Toil :).

File: cwltool/main.py (hunk 1 above)
Comment (mr-c) [reply cwltool/main.py:137]: Good point, I'll add a try/except

**Pull request merged**